js/app/utils.js

```diff
@@ -560,3 +561,34 @@ function intersection(a, b) {
         .forEach(item => intersection.add(item));
     return [...intersection];
 }
+
+function parse_uri (uri_str, parse_query) {
+    let uri = new SoupURI (uri_str);
```

Comment:
This seems like it should not work - isn't it `SoupURI.new(uri_str)`? `new SoupURI` would be calling `g_object_new()`.

Reply:
SoupURI is not a GObject, so I assume soup_uri_new() is mapped as the constructor of the boxed type